plugins/Compressor/CompressorControlDialog.h

```diff
@@ -0,0 +1,230 @@
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+#ifndef COMPRESSOR_CONTROL_DIALOG_H
+#define COMPRESSOR_CONTROL_DIALOG_H
+
+#include "EffectControlDialog.h"
+#include <QLabel>
+#include <QBasicTimer>
+#include "Knob.h"
+#include "../Eq/EqFader.h"
+#include "MainWindow.h"
+#include "GuiApplication.h"
+#include "PixmapButton.h"
+#include <QMouseEvent>
+#include <QTime>
+
+
+constexpr int COMP_MILLI_PER_PIXEL = 6;
+constexpr int MIN_COMP_SCREEN_X = 800;
+constexpr int MIN_COMP_SCREEN_Y = 360;
+constexpr int MAX_COMP_SCREEN_X = 1920;
+constexpr int MAX_COMP_SCREEN_Y = 1080;
+constexpr int COMP_SCREEN_X = 800;
+constexpr int COMP_SCREEN_Y = 560;
+constexpr int KNEE_SCREEN_X = COMP_SCREEN_Y;
+constexpr int KNEE_SCREEN_Y = COMP_SCREEN_Y;
+constexpr int COMP_KNEE_LINES = 20;
+constexpr int COMP_BOX_X = 720;
+constexpr int COMP_BOX_Y = 280;
+
+constexpr float COMP_NOISE_FLOOR = 0.000001;// -120 dbFs
+
+
+
+class CompressorControls;
+
+class CompressorControlDialog : public EffectControlDialog
+{
+	Q_OBJECT
+public:
+	CompressorControlDialog(CompressorControls* controls);
+
+	bool isResizable() const override {return true;}
+	QSize sizeHint() const override {return QSize(COMP_SCREEN_X, COMP_SCREEN_Y);}
+
+	// For theming purposes
+	Q_PROPERTY(QColor inVolAreaColor READ inVolAreaColor WRITE setInVolAreaColor)
+	Q_PROPERTY(QColor inVolColor READ inVolColor WRITE setInVolColor)
+	Q_PROPERTY(QColor outVolAreaColor READ outVolAreaColor WRITE setOutVolAreaColor)
+	Q_PROPERTY(QColor outVolColor READ outVolColor WRITE setOutVolColor)
+	Q_PROPERTY(QColor gainReductionColor READ gainReductionColor WRITE setGainReductionColor)
+	Q_PROPERTY(QColor kneeColor READ kneeColor WRITE setKneeColor)
+	Q_PROPERTY(QColor kneeColor2 READ kneeColor2 WRITE setKneeColor2)
+	Q_PROPERTY(QColor threshColor READ threshColor WRITE setThreshColor)
+	Q_PROPERTY(QColor textColor READ textColor WRITE setTextColor)
+	Q_PROPERTY(QColor graphColor READ graphColor WRITE setGraphColor)
+	Q_PROPERTY(QColor resetColor READ resetColor WRITE setResetColor)
+
+	QColor const & inVolAreaColor() const;
+	void setInVolAreaColor( const QColor & c );
+
+	QColor const & inVolColor() const;
+	void setInVolColor( const QColor & c );
+
+	QColor const & outVolAreaColor() const;
+	void setOutVolAreaColor( const QColor & c );
+
+	QColor const & outVolColor() const;
+	void setOutVolColor( const QColor & c );
+
+	QColor const & gainReductionColor() const;
+	void setGainReductionColor( const QColor & c );
+
+	QColor const & kneeColor() const;
+	void setKneeColor( const QColor & c );
+
+	QColor const & kneeColor2() const;
+	void setKneeColor2( const QColor & c );
+
+	QColor const & threshColor() const;
+	void setThreshColor( const QColor & c );
+
+	QColor const & textColor() const;
+	void setTextColor( const QColor & c );
+
+	QColor const & graphColor() const;
+	void setGraphColor( const QColor & c );
+
+	QColor const & resetColor() const;
+	void setResetColor( const QColor & c );
```

Comment:
Instead of `READ ... WRITE ...` we can use `MEMBER m_variable` to auto-generate the getter/setter functions. Saves a lot of boilerplate repetition that just sets or returns values.